mlflow/sklearn/utils.py

```diff
@@ -310,7 +311,10 @@ def plot_confusion_matrix(*args, **kwargs):
                 "axes.labelsize": 10.0,
             }
         ):
-            return sklearn.metrics.plot_confusion_matrix(*args, **kwargs)
+            if Version(sklearn.__version__) >= Version("1.0"):
+                return ConfusionMatrixDisplay.from_predictions(*args, **kwargs)
+            else:
+                return sklearn.metrics.plot_confusion_matrix(*args, **kwargs)
```

Comment:
Thanks for the change! I think there are a few more plotting functions. Can you also fix them?

Reply:
```
2022/04/14 23:14:39 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/sklearn/utils/deprecation.py:87: FutureWarning: Function plot_roc_curve is deprecated; Function :func:`plot_roc_curve` is deprecated in 1.0 and will be removed in 1.2. Use one of the class methods: :meth:`sklearn.metric.RocCurveDisplay.from_predictions` or :meth:`sklearn.metric.RocCurveDisplay.from_estimator`."
2022/04/14 23:14:39 WARNING mlflow.utils.autologging_utils: MLflow autologging encountered a warning: "/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/sklearn/utils/deprecation.py:87: FutureWarning: Function plot_precision_recall_curve is deprecated; Function `plot_precision_recall_curve` is deprecated in 1.0 and will be removed in 1.2. Use one of the class methods: PrecisionRecallDisplay.from_predictions or PrecisionRecallDisplay.from_estimator."
2022/04/14 23:14:41 WARNING mlflow.sklearn.utils: _get_classifier_artifacts.<locals>.plot_confusion_matrix failed. The artifact val_confusion_matrix will not be recorded. Artifact error: from_predictions() got an unexpected keyword argument 'estimator'
```

Reply:
Or we can focus on `plot_confusion_matrix` in this PR, and then fix the rest of the warnings in a separate PR.

Reply:
@shaikmoeed Or we can fix `plot_confusion_matrix` first and then fix `plot_roc_curve` and `plot_precision_recall_curve` later.

Reply:
Fixed one by one warnings for `plot_confusion_matrix`, `plot_roc_curve` and `plot_precision_recall_curve` plotting functions.